File: 04-Secuencial/01-Destacado/05-Haskell/Destacado.hs
{-
   Programa para la empresa de seguridad Aros S.A.
   que permita resaltar el nombre del empleado destacado
   del mes ofreciendo unas felicitaciones públicas.
   Autor: Julián Esteban Gutiérrez Posada
   Fecha: Febrero 2025
   Licencia: GNU GPL v3
-}

import Text.Printf (printf)
import Modulo.Util (mostrarMensaje)

main :: IO()
main = do
  nombreEmpleado <- ingresarTexto "Nombre del empleado destacado: "
  nombreMes <- ingresarTexto "Nombre del mes: "
  let felicitaciones = generarFelicitaciones nombreEmpleado nombreMes
  mostrarMensaje felicitaciones

ingresarTexto :: String -> IO String
ingresarTexto pregunta = do
  mostrarMensaje pregunta
  getLine

generarFelicitaciones :: String -> String -> String
generarFelicitaciones nombreEmpleado nombreMes =
  printf
    ("\nLa empresa de seguridad Aros S.A. quiere felicitar" ++
    "\npúblicamente a %s como nuestro" ++
    "\nempleado destacado del mes de %s," ++
    "\nmuchas gracias por su excelencia.")
    nombreEmpleado
    nombreMes

File: 04-Secuencial/01-Destacado/01-Python/Destacado.py
"""
    Programa para la empresa de seguridad Aros S.A.
    que permita resaltar el nombre del empleado destacado 
    del mes ofreciendo unas felicitaciones públicas.
    Autor: Julián Esteban Gutiérrez Posada
    Fecha: Febrero 2025
    Licencia: GNU GPL v3
"""

from modulo.util import mostrar_mensaje

def main():
    nombre_empleado = ingresar_texto("Nombre del empleado destacado: ")
    nombre_mes = ingresar_texto("Nombre del mes: ")
    felicitaciones = generar_felicitaciones(nombre_empleado, nombre_mes)
    mostrar_mensaje(felicitaciones)

def ingresar_texto(pregunta):
    return input(pregunta)

def generar_felicitaciones(nombre_empleado, nombre_mes):
    return (
        f"\nLa empresa de seguridad Aros S.A. quiere felicitar"
        f"\npúblicamente a {nombre_empleado} como nuestro"
        f"\nempleado destacado del mes de {nombre_mes},"
        f"\nmuchas gracias por su excelencia."
    )

main()